Mobile/lib/Components/TaskEdit.dart

```diff
@@ -0,0 +1,281 @@
+
+    if(newDateTime != null) {
+      setState(() {
+        endDate = newDateTime;
+        _endDatetimestampController.text = newDateTime.toString();
+      });
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _appState = context.watch<RootAppState>();
+    return AlertDialog(
+      scrollable: true,
+      title: Row(
+        children: [
+          Text('Edit task'),
+          Spacer(),
+          TextButton(
+            onPressed: _deleteTaskConfimation,
+            child: Text('Delete'),
+          ),
+        ],
+      ), 
+      // Text('Edit task'),
+      actions: [
+        TextButton(
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+          child: Text('Cancel'),
+        ),
+        TextButton(
+          onPressed: _updateTask,
+          child: Text('Save'),
+        ),
+      ],
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Form(
+            key: _formKey,
+            child: Column(
+              children: [
+                TextFormField(
+                  initialValue: title,
+                  maxLength: 100,
+                  autovalidateMode: AutovalidateMode.onUnfocus,
+                  decoration: InputDecoration(labelText: 'Title'),
+                  onChanged: (value) => title = value,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Field is required and cannot be empty';
+                    }
+                    return null;
+                  },
+                ),
+                SizedBox(height: 16),
+                TextFormField(
+                  initialValue: description,
+                  maxLines: 3,
+                  maxLength: 255,
+                  autovalidateMode: AutovalidateMode.onUnfocus,
+                  decoration: InputDecoration(labelText: 'Description'),
+                  onChanged: (value) => description = value,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Field is required and cannot be empty';
+                    }
+                    return null;
+                  },
+                ),
+                SizedBox(height: 16),
+                TextFormField(
+                  initialValue: reward,
+                  keyboardType: TextInputType.number,
+                  autovalidateMode: AutovalidateMode.onUnfocus,
+                  decoration: InputDecoration(labelText: 'Reward'),
+                  onChanged: (value) => reward = value,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Field is required and cannot be empty';
+                    }
+      
+                    if (double.tryParse(value) == null) {
+                      return 'Please enter a hold number';
```

Comment:
It should be either `whole number` or `integer`
```suggestion
                      return 'Please enter a whole number';
```

Reply:
nice i didn't know how to spell that 😄